Update MSI documentation

codebrain opened the pull request:
Updates images for 6.3 (and future) installers.

codebrain committed:
Update MSI documentation

elasticmachine commented:
Pinging @elastic/es-core-infra

russcam reviewed (commented):
I can't comment inline on `docs/reference/images/msi_installer/msi_installer_xpack.png`, so will add here. This image needs to be updated to the newer UI that removes mention of the Basic license being valid only for 30 days. the Basic license applied is now perpetual in 6.3.0+, in line with other X-Pack bundled distributions

russcam reviewed (commented):
Line 406 of windows.asciidoc: replace use of `{es}` with `Elasticsearch`

russcam commented:
Replace usage of `{xpack}` with X-Pack

russcam commented:
The warning section:

> WARNING: Uninstallation will remove **all** directories and their contents created as part of
installation, **including data within the data directory**. If you wish to retain your data upon
uninstallation, it is recommended that you make a copy of the data directory beforehand or consider using the snapshot API.

is out of date. [Since 6.2.0, the installer does **not** delete data, config or logs directories upon uninstall.](https://github.com/elastic/windows-installers/commit/7de7131df97a2cadf44605f6189a102ab9cb743e)

russcam commented:
The note section:

> NOTE: {xpack} includes a choice of a Trial or Basic license for 30 days. After that, you can obtain one of the
https://www.elastic.co/subscriptions[available subscriptions] or {ref}/security-settings.html[disable Security].
The Basic license is free and includes the https://www.elastic.co/products/x-pack/monitoring[Monitoring] extension.

should indicate that **only** the trial license is valid for 30 days; the Basic license is perpetual.

russcam commented:
Example code

```
start /wait msiexec.exe /i elasticsearch-{version}.msi /qn INSTALLDIR="C:\Custom Install Directory" PLUGINS="x-pack"
```

must include the version as the final directory. This can be achieved by using the `{version}` attribute and [allowing attribute substitutions in the `source` block](https://stackoverflow.com/questions/29124154/substitutions-inside-literals-in-asciidoc)

codebrain commented:
@russcam 

1. Have updated image to reflect new license text.
2. Replaced all {es} and {xpack}.
3. Warning section updated to reflect data, config or logs directories not being deleted.
4. License validity period updated.
5. Added {version} to install directory via command line.

codebrain committed:
Address PR comments

russcam commented on docs/reference/setup/install/windows.asciidoc:
```diff
@@ -467,13 +468,13 @@ start /wait msiexec.exe /i elasticsearch-{version}.msi /qn /l upgrade.log
-installation, **including data within the data directory**. If you wish to retain your data upon
-uninstallation, it is recommended that you make a copy of the data directory beforehand or consider using the snapshot API.
+installation, **except for data, config or logs directories**. If you wish to retain your data upon
+uninstallation, it is recommended that you make a copy of your data directory beforehand or consider using the snapshot API.
```
> If you wish to retain your data upon uninstallation, it is recommended that you make a copy of your data directory

This doesn't align with the previous sentence, since data directory is not removed on uninstall

codebrain replied:
Yes. I think asking the user to backup their data on an upgrade is probably good advice anyway, so will rephrase.

russcam commented on docs/reference/setup/install/windows.asciidoc:
```diff
@@ -76,12 +76,12 @@ TIP: Ensure the installation machine has access to the internet and that any cor
 image::images/msi_installer/msi_installer_xpack.png[]
 
-NOTE: {xpack} includes a choice of a Trial or Basic license for 30 days. After that, you can obtain one of the
+NOTE: X-Pack includes a choice of a Trial license (valid for 30 days) or Basic license (valid perpetually). After that, you can obtain one of the
```
Might be a bit clearer to separate out the license types from their validity e.g.

X-Pack includes a choice of a Trial or Basic license. A Trial license is valid for 30 days, after which you can obtain one of  the available subscriptions. The Basic license on the other hand is free and perpetual. Consult the https://www.elastic.co/subscriptions[available subscriptions] for further details on which features are available under which license.

russcam reviewed (commented):
Left a couple more comments.

I'm wondering if the term X-Pack should be de-emphasised here too. @eskibars, thoughts?

codebrain committed:
Address PR comments

russcam reviewed (approved):
Changes LGTM. I think we can always follow up re: X-Pack in a subsequent PR.

codebrain commented:
Ported to `6.x`, `6.3`, `6.4` and `master`
